cli/crawl-cli.js

```diff
@@ -95,35 +100,49 @@ async function run(inputUrls, outputPath, verbose, logPath, numberOfCrawlers, da
         return true;
     });
 
-    // show progress bar only if we are not printing all logs to screen (verbose)
-    const progressBar = (verbose || urls.length === 0) ? null : new ProgressBar('[:bar] :percent ETA :etas fail :fail% :site', {
-        complete: chalk.green('='),
-        incomplete: ' ',
-        total: urls.length,
-        width: 30
-    });
-
+    const urlsLength = urls.length;
     let failures = 0;
     let successes = 0;
+
+    /**
+     * @type {Error}
+     */
+    let fatalError = null;
+
+    /**
+     * @type {Array<Array<number>>}
+     */
+    let crawlTimes = [];
+    
     // eslint-disable-next-line arrow-parens
-    const updateProgress = (/** @type {string} */site = '') => {
-        if(progressBar) {
-            progressBar.tick({
-                site,
-                fail: (failures / (failures + successes) * 100).toFixed(1)
-            });
-        }
+    const updateProgress = (/** @type {string} */site = '', /** @type {{testStarted: number, testFinished: number, data: {screenshots: string}}}} */data) => {
+        reporters.forEach(reporter => {
+            reporter.update({site, successes, failures, urls: urlsLength, data, crawlTimes, fatalError, numberOfCrawlers, regionCode});
+        });
     };
 
     /**
      * @param {URL} url
-     * @param {object} data
+     * @param {{testStarted: number, testFinished: number, data: {screenshots: string}}} data
      */
     const dataCallback = (url, data) => {
         successes++;
-        updateProgress(url.toString());
+
+        crawlTimes.push([data.testStarted, data.testFinished, data.testFinished - data.testStarted]);
 
         const outputFile = createOutputPath(url);
+
+        // temp name for the screenshot is scored in data. rename the screenshot to match the file crawl file
```

Comment:
I think this comment is now out of date (screenshot is not renamed)

Reply:
ah, good call, I'll fix that. Thanks for reviewing!